src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1197 @@
+			{
+				ResetControlWidth();
+			}
+		}
+
+		private void UpdateRatingItemsAppearance()
+		{
+			if (m_foregroundStackPanel != null)
+			{
+				// TODO: MSFT 11521414 - complete disabled state functionality
+
+				double placeholderValue = PlaceholderValue;
+				double ratingValue = Value;
+				double value = 0.0;
+
+				if (m_isPointerOver)
+				{
+					value = Math.Ceiling(m_mousePercentage * MaxRating);
+					if (ratingValue == c_noValueSetSentinel)
+					{
+						if (placeholderValue == -1)
+						{
+							VisualStateManager.GoToState(this, "PointerOverPlaceholder", false);
+							CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.PointerOverPlaceholder);
+						}
+						else
+						{
+							VisualStateManager.GoToState(this, "PointerOverUnselected", false);
+							// The API is locked, so we can't change this part to be consistent any more:
+							CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.PointerOverPlaceholder);
+						}
+					}
+					else
+					{
+						VisualStateManager.GoToState(this, "PointerOverSet", false);
+						CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.PointerOverSet);
+					}
+				}
+				else if (ratingValue > c_noValueSetSentinel)
+				{
+					value = ratingValue;
+					VisualStateManager.GoToState(this, "Set", false);
+					CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.Set);
+				}
+				else if (placeholderValue > c_noValueSetSentinel)
+				{
+					value = placeholderValue;
+					VisualStateManager.GoToState(this, "Placeholder", false);
+					CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.Placeholder);
+				} // there's no "unset" state because the foreground items are simply cropped out
+
+				if (!IsEnabled)
+				{
+					// TODO: MSFT 11521414 - complete disabled state functionality [merge this code block with ifs above]
+					VisualStateManager.GoToState(this, "Disabled", false);
+					CustomizeStackPanel(m_foregroundStackPanel, RatingControlStates.Disabled);
+				}
+
+				uint i = 0;
+				foreach (var uiElement in m_foregroundStackPanel.Children)
+				{
+					// Handle clips on stars
+					float width = RenderingRatingFontSize();
+					if (i + 1 > value)
+					{
+						if (i < value)
+						{
+							// partial stars
+							width *= (float)(value - Math.Floor(value));
+						}
+						else
+						{
+							// empty stars
+							width = 0.0f;
+						}
+					}
+
+					Rect rect = new Rect();
+					rect.X = 0;
+					rect.Y = 0;
+					rect.Height = RenderingRatingFontSize();
+					rect.Width = width;
+
+					RectangleGeometry rg = new RectangleGeometry();
+					rg.Rect = rect;
+					uiElement.Clip = rg;
+
+					i++;
+				}
+
+				ResetControlWidth();
+			}
+		}
+
+		private void ApplyScaleExpressionAnimation(UIElement uiElement, int starIndex)
+		{
+			//TODO: Uno Specific - Expression animations are not supported yet
+
+			//Visual uiElementVisual = ElementCompositionPreview.GetElementVisual(uiElement);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)